code/crates/engine/src/wal/entry.rs

```diff
@@ -120,7 +120,6 @@ fn encode_timeout(tag: u8, timeout: &Timeout, mut buf: impl Write) -> io::Result
         TimeoutKind::Propose => 1,
         TimeoutKind::Prevote => 2,
         TimeoutKind::Precommit => 3,
-        TimeoutKind::Commit => 4,
```

Comment:
Note that with removing of `TimeoutKind::Commit`, we are now skipping the encoded value `4`. Not sure this is confusing, let me know if I should adjust to avoid skipping 4

Reply:
We should probably adjust the code to skip entries with tag `4` in order to maintain backward compatibility and preserve the ability to replay a WAL containing commit timeouts.

Reply:
Actually that's fine because we already just skip entries that fail to decode. But maybe we can yield a specific error to be explicit about commit timeouts having been removed.

Reply:
4f1166b0e157e4e8e30a4c2f9674b58743d8e593